From 6facfb8ddfcd2a85565230f511994b01fc7883f8 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Tue, 15 Sep 2015 15:50:18 -0500
Subject: [PATCH 0717/5242] MLK-11555: mtd: nand: i.MX6UL supports bitflip
 detection

commit  aed69c8e3997c52e07a40debc62cf77314c79d84 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX6UL also has the DEBUG1 register which can be used for bitflip
detection for erased page.

Signed-off-by: Han Xu <b45815@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c  |    4 ++--
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    3 ++-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
index 28f4a3c..8cb2586 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-lib.c
@@ -305,8 +305,8 @@ int bch_set_geometry(struct gpmi_nand_data *this)
 			| BF_BCH_FLASH0LAYOUT1_DATAN_SIZE(block_size, this),
 			r->bch_regs + HW_BCH_FLASH0LAYOUT1);
 
-	/* Set erase threshold to gf/2 for mx6qp and mx7 */
-	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7D(this))
+	/* Set erase threshold to gf/2 for mx6ul, mx6qp and mx7 */
+	if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7D(this) || GPMI_IS_MX6UL(this))
 		writel(BF_BCH_MODE_ERASE_THRESHOLD(gf_len/2),
 			r->bch_regs + HW_BCH_MODE);
 
diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index f413ec9..04064f9 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -1040,7 +1040,8 @@ static int gpmi_ecc_read_page_data(struct nand_chip *chip,
 			continue;
 
 		if (*status == STATUS_ERASED) {
-			if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7D(this))
+			if (GPMI_IS_MX6QP(this) || GPMI_IS_MX7D(this) ||
+						GPMI_IS_MX6UL(this))
 				if (readl(bch_regs + HW_BCH_DEBUG1))
 					flag = 1;
 			continue;
-- 
1.7.9.5

